// @angular/forms/src/directives/default_value_accessor.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register(['@angular/forms/src/directives/default_value_accessor'], function(exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var import0;
    var Wrapper_DefaultValueAccessor;
    return {
        setters:[
            function (import0_1) {
                import0 = import0_1;
            }],
        execute: function() {
            Wrapper_DefaultValueAccessor = (function () {
                function Wrapper_DefaultValueAccessor(p0, p1) {
                    this._changed = false;
                    this.context = new import0.DefaultValueAccessor(p0, p1);
                }
                Wrapper_DefaultValueAccessor.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_DefaultValueAccessor.prototype.ngOnDestroy = function () {
                };
                Wrapper_DefaultValueAccessor.prototype.ngDoCheck = function (view, el) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_DefaultValueAccessor.prototype.checkHost = function (view, componentView, el) {
                };
                Wrapper_DefaultValueAccessor.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    if ((eventName == 'input')) {
                        var pd_sub_0 = (this.context.onChange($event.target.value) !== false);
                        result = (pd_sub_0 && result);
                    }
                    if ((eventName == 'blur')) {
                        var pd_sub_1 = (this.context.onTouched() !== false);
                        result = (pd_sub_1 && result);
                    }
                    return result;
                };
                Wrapper_DefaultValueAccessor.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_DefaultValueAccessor;
            }());
            exports_1("Wrapper_DefaultValueAccessor", Wrapper_DefaultValueAccessor);
        }
    }
});
